arangod/Agency/Agent.cpp

```diff
@@ -1519,97 +1519,107 @@ void Agent::triggerPollsNoLock(query_t qu, SteadyTimePoint const& tp) {
 
 /// Send out append entries to followers regularly or on event
 void Agent::run() {
-  // Only run in case we are in multi-host mode
-  while (!this->isStopping() && size() > 1) {
-    {
-      // We set the variable to false here, if any change happens during
-      // or after the calls in this loop, this will be set to true to
-      // indicate no sleeping. Any change will happen under the mutex.
-      CONDITION_LOCKER(guard, _appendCV);
-      _agentNeedsWakeup = false;
+
+  if (size() == 1) {
```

Comment:
Let's add a `try {} catch () {}` here and in the multi-agent case to protect the thread against exceptions.